src/Splat/Platforms/Cocoa/Bitmaps/PlatformBitmapLoader.cs

```diff
@@ -30,43 +30,59 @@ public Task<IBitmap> Load(Stream sourceStream, float? desiredWidth, float? desir
             var data = NSData.FromStream(sourceStream);
 
             var tcs = new TaskCompletionSource<IBitmap>();
-            NSRunLoop.Main.BeginInvokeOnMainThread(() =>
+#if UIKIT
+            NSRunLoop.InvokeInBackground(() =>
             {
                 try
                 {
-#if UIKIT
                     var bitmap = UIImage.LoadFromData(data);
                     if (bitmap == null)
                     {
                         throw new Exception("Failed to load image");
                     }
 
                     tcs.TrySetResult(new CocoaBitmap(bitmap));
-#else
-                    tcs.TrySetResult(new CocoaBitmap(new UIImage(data)));
-#endif
                 }
                 catch (Exception ex)
                 {
                     tcs.TrySetException(ex);
                 }
             });
+#else
+            tcs.TrySetResult(new CocoaBitmap(new UIImage(data)));
```

Comment:
Does this require a Main.BeginInvokeOnMainThread(()?

Reply:
Not 100% sure if it requires a BeginInvokeOnMainThread or not. At the time I assumed not since it’s just creating objects.

Yeah, I think I’ve seen in another library where they just use Task.Run.